src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -109,8 +112,8 @@ public function validate(mixed $value, Constraint $constraint): void
             return;
         }
 
-        // should contain uppercase characters only
-        if (strtoupper($canonicalize) !== $canonicalize) {
+        // should contain uppercase characters only in strict mode
+        if (Bic::VALIDATION_MODE_STRICT === $constraint->mode && strtoupper($canonicalize) !== $canonicalize) {
```

Comment:
@xabbuh I changed it now to mode, the way that was also done in the email validator, because i need id here for the check. With the normalizer way i wouldn't know to check this or not.